### PR TITLE
Hotfix/validate app cert

### PR DESCRIPTION
**Pull request by cristipogacean**
* fix for #430
* default certificate store to Directory for windows containers

### CODE REVIEW

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Gateway/src/Server/GatewayServer.cs [hunk 1]
```diff
@@ -1608,25 +1608,25 @@ public sealed class GatewayServer : SessionServerBase, IServer {
...
+                        StoreType = CertificateStoreType.Directory,
                         StorePath = "pki/issuers",
                     },
                     RejectedCertificateStore = new CertificateTrustList {
-                        StoreType = "Directory",
+                        StoreType = CertificateStoreType.Directory,
                         StorePath = "pki/rejected"
                     },
+                    MinimumCertificateKeySize = 1024,
+                    RejectSHA1SignedCertificates = true,
```
Comment (marcschier) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Gateway/src/Server/GatewayServer.cs:1628]: We should allow sha1 for interop.  Let's add a workitem to warn user in the new cert validation API.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/Default/ClientServicesConfig.cs [hunk 2]
```diff
@@ -63,11 +62,12 @@ public class ClientServicesConfig : ConfigBase, IClientServicesConfig, ISecurity
         /// <inheritdoc/>
         public bool RejectSha1SignedCertificates => _security.RejectSha1SignedCertificates;
```
Comment (marcschier) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/Default/ClientServicesConfig.cs:63]: One could use this option.  But I would default to false and as per discussion minimize config surface.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/ISecurityConfig.cs [hunk 3]
```diff
@@ -50,5 +50,10 @@ public interface ISecurityConfig {
         /// Trusted peer certificates
         /// </summary>
         CertificateStore TrustedPeerCertificates { get; }
+
+        /// <summary>
+        /// Automatically add application certificate to the trusted store
+        /// </summary>
+        public bool AddAppCertToTrustedStore { get; }
```
Comment (marcschier) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/ISecurityConfig.cs:57]: Public is not needed.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Gateway/src/Server/GatewayServer.cs (hunk 1 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Gateway/src/Server/GatewayServer.cs:1628]: done, DevOps backlog item created as well

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/Default/ClientServicesConfig.cs (hunk 2 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/Default/ClientServicesConfig.cs:63]: it is defaulted to false, anywhere except Gateway Server, (previous comment,), where was already set to true and did not questioned the reason :)

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/ISecurityConfig.cs (hunk 3 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/ISecurityConfig.cs:57]: totally! changed

**Pull request merged**